x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/dimension_panel.test.tsx

```diff
@@ -276,31 +231,90 @@ describe('IndexPatternDimensionPanel', () => {
             },
           },
         }}
-        setState={setState}
-        columnId={'col1'}
-        filterOperations={() => true}
       />
     );
 
-    wrapper
-      .find('[data-test-subj="indexPattern-dimensionPopover-button"]')
-      .first()
-      .simulate('click');
+    openPopover();
 
-    expect(wrapper.find(EuiContextMenuItem).map(instance => instance.text())).toEqual([
-      'Minimum',
-      'Maximum',
-      'Average',
-      'Sum',
-    ]);
+    const options = (wrapper.find(EuiSideNav).prop('items')[0].items as unknown) as Array<{
+      name: string;
+      className: string;
+    }>;
+
+    expect(options.find(({ name }) => name === 'Minimum')!.className).not.toContain('incompatible');
+
+    expect(options.find(({ name }) => name === 'Date Histogram')!.className).toContain(
+      'incompatible'
+    );
+  });
+
+  it('should keep the operation when switching to another field compatible with this operation', () => {
+    const initialState: IndexPatternPrivateState = {
+      ...state,
+      columns: {
+        ...state.columns,
+        col1: {
+          operationId: 'op1',
+          label: 'Max of bytes',
+          dataType: 'number',
+          isBucketed: false,
+
+          // Private
+          operationType: 'max',
+          sourceField: 'bytes',
+        },
+      },
+    };
+
+    wrapper = mount(<IndexPatternDimensionPanel {...defaultProps} state={initialState} />);
+
+    openPopover();
+
+    const comboBox = wrapper.find(EuiComboBox)!;
+    const option = comboBox.prop('options')![1].options!.find(({ label }) => label === 'memory')!;
+
+    comboBox.prop('onChange')!([option]);
+
+    expect(setState).toHaveBeenCalledWith({
+      ...initialState,
+      columns: {
+        ...state.columns,
+        col1: expect.objectContaining({
+          operationType: 'max',
+          sourceField: 'memory',
+          // Other parts of this don't matter for this test
+        }),
+      },
+    });
   });
 
-  it('should update the datasource state on selection of an operation', () => {
-    const setState = jest.fn();
+  it('should allow switching to incompatible fields for the current operation and switch to a compatible one', () => {
```

Comment:
This test description is not very clear to me. I think I would write it as `it('should switch operations when selecting a field that requires another operation', () => {`

Reply:
Way better, thanks!